docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-google-cloud-functions.md

```diff
@@ -0,0 +1,115 @@
+
+To deploy a pipeline using the GCP Cloud functions, you'll first need to set up a blank repo in Cloud Source Repositories, a service provided by GCP for hosting repositories, or you can clone it to your local machine and then install the Google Cloud SDK on your local machine.
+
+## 1. Setup Pipeline in Google Cloud Repositories
+To deploy the pipeline, we'll use the Google Cloud Source Repositories(first method):
+
+1. Sign in to your GCP account and enable the Cloud Functions API.
+2. To set up the environment, you can follow these steps:
+    - Create an empty repo in Cloud Source Repositories.
+      (Cloud Source repository is a service in GCP for hosting repositories from different platforms).
+        
+    - After making the repo, click Edit repo to open it in a "Shell Editor".
+3. In this guide, we'll demonstrate using the `dlt` [Notion verified source](https://dlthub.com/docs/dlt-ecosystem/verified-sources/notion). However, you can use any verified source or create a custom one to suit your needs.
+4. In the shell editor
+    - Run the following command to initialise the verified source with Notion and create a pipeline with BigQuery as the target.
+    
+      ```bash
+      dlt init notion bigquery
+      ```
+    
+    - After running the command, a new directory will be created with the necessary files and configurations.
+    - Detailed information about initialising a verified source and a pipeline example can be found in the `dlthub` [documentation](https://dlthub.com/docs/dlt-ecosystem/verified-sources/notion).
+5. Rename the file "notion_pipeline.py" to "main.py" and modify the code as follows:
+    
+    ```python
+    import os
+    import dlt
+    from .notion import notion_databases
+    
+    def load_databases() -> str:
+        pipeline = dlt.pipeline(
+            pipeline_name="notion",
+            destination='bigquery',
+            dataset_name="notion_data_aman_atul",
+        )
+        data = notion_databases()
+    
+        info = pipeline.run(data)
+        print(info)
+        return "Data loaded successfully". # returns the value
+    
+    # Created a new function to call the load_database function 
+    def pipeline_notion(request):
+        return load_databases()
+    ```
+    In this code, we define a function named "pipeline_notion" that will be invoked by the cloud function. By default, Google Cloud Functions searches for the function in "main.py".
+
+6. Further, you need to modify,  the `__init__.py` file in the notion folder, and modify the `notion_databases` function as follows:
+    
+    ```python
+    @dlt.source
+    def notion_databases(
+        database_ids: Optional[List[Dict[str, str]]] = None,
+        api_key = notion_apikey_env, # Refers to the notion secret
+    ) -> Iterator[DltResource]:
+    ```
+    
+    Note how we have referenced api_key to the variable named "notion_apikey_env" instead of "dlt.secrets.value".
+    ### Managing secrets with Secret Manager
```

Comment:
this is only necessary when secrets manager is used. you can just use ENV Variables. see my other comments

Reply:
deleted